src/main/java/org/opentripplanner/updater/configure/UpdaterConfigurator.java

```diff
@@ -137,7 +141,7 @@ private List<GraphUpdater> createUpdatersFromConfig() {
 
     if (!updatersParameters.getVehicleRentalParameters().isEmpty()) {
       int maxHttpConnections = updatersParameters.getVehicleRentalParameters().size();
-      OtpHttpClient otpHttpClient = new OtpHttpClient(maxHttpConnections);
+      OtpHttpClient otpHttpClient = new OtpHttpClient(maxHttpConnections, LOG);
```

Comment:
This is the clearest example of problems with passing in the logger while constructing the client. I haven't looked into if we could avoid these types of problems if we pass in the logger to the appropriate methods instead. Not sure which approach is better.

Reply:
I've created a factor for OtpHttpClient which gets rid of this issue.